src/main/pileup.js

```diff
@@ -80,6 +80,37 @@ function create(elOrId: string|Element, params: PileupParams): Pileup {
       ReactDOM.render(<Root referenceSource={referenceTrack.source}
                             tracks={vizTracks}
                             initialRange={params.range} />, el);
+
+  //if the element doesn't belong to document DOM observe DOM to detect
+  //when it's attached
+  var observer = null;
+  if (!document.contains(el)) {
+    observer = new MutationObserver(function(mutations) {
+      mutations.forEach(function(mutation) {
+        if (mutation.type === 'childList') {
+          var added= false;
+          for (var i=0; i<mutation.addedNodes.length;i++) {
+            if (mutation.addedNodes[i]===el) {
+              added = true;
+            }
+          }
+          if (added) {
+            if (reactElement) {
+              reactElement.setState({updateSize:true});
+            } else {
+              throw 'ReactElement was not initialized properly';
```

Comment:
Well, I don't like when the code fails to do what it's supposed to do and don't give any hint what went wrong. 
It helps debugging, testing etc.